crane/models.py

```diff
@@ -23,8 +23,6 @@ def load_from_settings(self, settings):
         self.stack = rancher.Stack.from_name(settings['stack'])
         self.services = [self.stack.service_from_name(service) for service in settings['service']]
 
-        self.repo = git.Repo(environ['CI_PROJECT_DIR'])
```

Comment:
Don't remove this, we want a predictable set of attributes on this class, `repo` should always be available, even if it's just set to `None`.

Since you'll set repo in `__init__`, you might as well move the error handling here.

Reply:
isn't it already set on L16? 🤔 

```
    repo = attr.ib(default=None)
```

Reply:
Oh that's right, my bad.